docs/cudf/source/developer_guide/library_design.md

```diff
@@ -314,3 +315,190 @@ The pandas API also includes a number of helper objects, such as `GroupBy`, `Rol
 cuDF implements corresponding objects with the same APIs.
 Internally, these objects typically interact with cuDF objects at the Frame layer via composition.
 However, for performance reasons they frequently access internal attributes and methods of `Frame` and its subclasses.
+
+
+(copy-on-write-dev-doc)=
+
+## Copy-on-write
+
+This section describes the internal implementation details of the copy-on-write feature.
+It is recommended that developers familiarize themselves with [the user-facing documentation](copy-on-write-user-doc) of this functionality before reading through the internals
+below.
+
+The core copy-on-write implementation relies on the `CopyOnWriteBuffer` class.
```

Comment:
I think we did just above no? (`(copy-on-write-user-doc)`)

Reply:
I think @wence- wants a link to the class, something like :class:`cudf.core.buffer.CopyOnWriteBuffer`. AFAIK buffer classes don't actually show up in our rendered docs though since they aren't public APIs, so we may not benefit from this.

Reply:
Right, we don't generate docstrings for non-public APIs, so we will not really see a link here anyway.